pkg/auth/setup.go

```diff
@@ -47,6 +52,85 @@ func attachPolicies(ctx context.Context, authService Service, groupID string, po
 	return nil
 }
 
+// statementForPolicyType holds the Statement for a policy by its name,
+// without the required ARN.
+var statementByName = map[string]model.Statement{
+	"AllAccess": {
+		Action: []string{"fs:*", "auth:*", "ci:*", "retention:*", "branches:*"},
+		Effect: model.StatementEffectAllow,
+	},
+	"FSFullAccess": {
+		Action: []string{
+			"fs:*",
+		},
+		Effect: model.StatementEffectAllow,
+	},
+	"FSReadWrite": {
+		Action: []string{
+			permissions.ListRepositoriesAction,
+			permissions.ReadRepositoryAction,
+			permissions.ReadCommitAction,
+			permissions.ListBranchesAction,
+			permissions.ListTagsAction,
+			permissions.ListObjectsAction,
+			permissions.ReadObjectAction,
+			permissions.WriteObjectAction,
+			permissions.DeleteObjectAction,
+			permissions.RevertBranchAction,
+			permissions.ReadBranchAction,
+			permissions.ReadTagAction,
+			permissions.CreateBranchAction,
+			permissions.CreateTagAction,
+			permissions.DeleteBranchAction,
+			permissions.DeleteTagAction,
+			permissions.CreateCommitAction,
+		},
+		Effect: model.StatementEffectAllow,
+	},
+	"FSRead": {
+		Action: []string{
+			"fs:List*",
+			"fs:Read*",
+		},
+
+		Effect: model.StatementEffectAllow,
+	},
+	"AuthManageOwnCredentials": {
+		Action: []string{
+			permissions.CreateCredentialsAction,
+			permissions.DeleteCredentialsAction,
+			permissions.ListCredentialsAction,
+			permissions.ReadCredentialsAction,
+		},
+		Effect: model.StatementEffectAllow,
+	},
+}
+
+// MakeStatementForPolicyType returns statements for policy type typ,
+// limited to resources.
+func MakeStatementForPolicyType(typ string, resources []string) (model.Statements, error) {
+	statement, ok := statementByName[typ]
+	if !ok {
+		return nil, fmt.Errorf("%w: %s", ErrStatementNotFound, typ)
+	}
+	statements := make(model.Statements, len(resources))
+	for i, resource := range resources {
+		if statements[i].Resource == "" {
+			statements[i] = statement
+			statements[i].Resource = resource
+		}
+	}
+	return statements, nil
+}
+
+func MakeStatementForPolicyTypeOrDie(typ string, resources []string) model.Statements {
```

Comment:
Suggestion: change `typ` from string to type

Reply:
Not done: this is a word in the GUI.  In general, the raw string / named string type distinction in package auth is unclear.